eureka-client/src/main/java/com/netflix/discovery/endpoint/DnsResolver.java

```diff
@@ -33,15 +33,10 @@
     private static final String CNAME_RECORD_TYPE = "CNAME";
     private static final String TXT_RECORD_TYPE = "TXT";
 
-    static final DirContext dirContext = getDirContext();
-
-    private DnsResolver() {
-    }
-
     /**
      * Load up the DNS JNDI context provider.
      */
-    public static DirContext getDirContext() {
+    private DirContext getDirContext() {
```

Comment:
We can see that this method can `throw new RuntimeException`. It's okay to throw this when application starts: this is a critical problem which has rights to prevent application startup. But did you check what are the consequences of this happening at some point in time when application is already operating and just wants to refresh list of Eureka nodes?

Reply:
That's a good point. I have changed all invocations and ensure that this RuntimeException properly caught and logged.